src/github-client.ts

```diff
@@ -0,0 +1,37 @@
+export interface GitHubClient {
+  retryTimeout: number;
+  hasApiToken(): boolean;
+  getTags(page: number, limit: number): Promise<any>;
+}
+
+export class DefaultGitHubClient implements GitHubClient {
+  retryTimeout: number = 5000;
+  private githubToken: string | null;
+
+  constructor(githubToken: string | null = null) {
+    this.githubToken = githubToken || process.env.GH_TOKEN || null;
+  }
+
+  hasApiToken(): boolean {
+    return this.githubToken != null && this.githubToken != "";
+  }
+
+  async getTags(page: number, limit: number): Promise<any> {
+    const url = `https://api.github.com/repos/swiftlang/swift/tags?per_page=${limit}&page=${page}`;
+    return await this.get(url);
+  }
+
+  private async get(url: string): Promise<any> {
+    let headers = {};
+    if (this.hasApiToken()) {
+      headers = {
+        Authorization: `Bearer ${this.githubToken}`,
+      };
+    }
+    const response = await fetch(url, {
+      headers: headers,
+    });
+    const json: any = await response.json();
+    return json;
+  }
+}
```

Comment:
Did you consider using the octokit and actions toolkit provided by GitHub to make requests?
(You can find an example of such an implementation here https://github.com/actions/toolkit/tree/main/packages/github)

Reply:
Will take a look, thanks for the reference!

Reply:
@fwal changed to octokit for requests